Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp

```diff
@@ -249,34 +258,79 @@ std::optional<BoundVector> estimateTrackParamsFromSeed(
   // Transform it back to the original frame
   Vector3 direction = rotation * transDirection.normalized();
 
-  // Initialize the bound parameters vector
-  BoundVector params = BoundVector::Zero();
+  // The estimated q/pt in [GeV/c]^-1 (note that the pt is the projection of
+  // momentum on the transverse plane of the new frame)
+  ActsScalar qOverPt = sign * (UnitConstants::m) / (0.3 * bFieldInTesla * R);
 
-  // The estimated phi and theta
-  params[eBoundPhi] = VectorHelpers::phi(direction);
-  params[eBoundTheta] = VectorHelpers::theta(direction);
+  Vector4 parameterOrigin(spGlobalPositions[0].x(), spGlobalPositions[0].y(),
+                          spGlobalPositions[0].z(),
+                          spGlobalTimes[0].value_or(0.));
 
   // Transform the bottom space point to local coordinates of the provided
   // surface
-  auto lpResult = surface.globalToLocal(gctx, spGlobalPositions[0], direction);
+  auto lpResult =
+      surface.globalToLocal(gctx, parameterOrigin.head<3>(), direction);
   if (!lpResult.ok()) {
-    ACTS_ERROR(
-        "Global to local transformation did not succeed. Please make sure the "
-        "bottom space point lies on the provided surface.");
-    return std::nullopt;
+    // no cov transport matrix is needed here
+    // particle hypothesis does not matter here
+    CurvilinearTrackParameters estimatedParams(parameterOrigin, direction,
+                                               qOverPt, std::nullopt,
+                                               ParticleHypothesis::pion());
+
+    auto surfaceIntersection =
+        surface.intersect(gctx, parameterOrigin.head<3>(), direction).closest();
+
+    if (!surfaceIntersection.isValid()) {
+      ACTS_INFO(
+          "The surface does not intersect with the origin and estimated "
+          "direction.");
+      return std::nullopt;
+    }
+
+    Direction propagatorDirection =
+        Direction::fromScalarZeroAsPositive(surfaceIntersection.pathLength());
+
+    Propagator propagator(
+        EigenStepper<>(std::make_shared<ConstantBField>(bField)),
```

Comment:
So I'm not a huge fan of allocating this shared magnetic field for all seeds...

Reply:
What about the idea returning the free parameters and leaving the propagation to the caller?

Reply:
I think we should either do the propagation in this function or remove the function as it is quite confusing with the surface as parameter. I agree with you that the magnetic field should rather be passed in. Better would even be to push the propagator in so the user can decide on which propagator will be used.

This is along the lines what @paulgessinger suggested. But if we add a propagator argument we will always require it since we cannot default construct it because of the field.

Reply:
Hmm so what I don't really like from an interface/design perspective, that this function can get silently very expensive with the propagation... 
I think the best solution would be to remove the surface from the interface, return curvilinear parameters, and provide a helper function for the propagation.

Reply:
I fear curvilinear will not really solve the problem because we use the parameters as starting parameters and then people might miss the measurement on the first surface. Curvilinear are also practically free parameters and we now have a function for that.

In that case I would maybe just drop the function and as you say provide a helper

Reply:
Ah right I forgot that we have the free parameters function now. Yeah, then this is the way to go I think.